drivers/overlay/ov_network.go

```diff
@@ -294,6 +295,12 @@ func (n *network) destroySandbox() {
 			}
 		}
 
+		// Close the netlink socket, this will also release the watchMiss goroutine that is using it
+		if n.nlSocket != nil {
+			n.nlSocket.Close()
```

Comment:
This needs locking to avoid a data race involving `nlSock.fd`. `Close` could race with `Receive` or `GetFd`.